File: app/src/main/java/ru/mrlargha/stemobile/database/DateConverter.java
package ru.mrlargha.stemobile.database;

import androidx.room.TypeConverter;

import java.util.Date;

@SuppressWarnings("WeakerAccess") // Type converters must be public
public class DateConverter {
    @TypeConverter
    public long fromSubstitutionDate(Date substitutionDate) {
        return substitutionDate.getTime();
    }

    @TypeConverter
    public Date toSubstitutionDate(long substitutionDate) {
        return new Date(substitutionDate);
    }
}

File: app/src/main/java/ru/mrlargha/stemobile/database/DateConverter.kt
package ru.mrlargha.stemobile.database

import androidx.room.TypeConverter
import java.util.*

// Type converters must be public
class DateConverter {
    @TypeConverter
    fun fromSubstitutionDate(substitutionDate: Date): Long {
        return substitutionDate.time
    }

    @TypeConverter
    fun toSubstitutionDate(substitutionDate: Long): Date {
        return Date(substitutionDate)
    }
}